packages/xml-views-completion/src/api.ts

```diff
@@ -24,9 +29,40 @@ export function getXMLViewCompletions(
   });
 
   const allowedVisibility: UI5Visibility[] = ["public", "protected"];
-  const publicAndProtectedSuggestions = filter(suggestions, _ =>
-    includes(allowedVisibility, _.ui5Node.visibility)
+  const publicAndProtectedSuggestions = filter(
+    suggestions,
+    _ =>
+      isLiteralXMLViewCompletion(_) ||
+      includes(allowedVisibility, _.ui5Node.visibility)
   );
 
   return publicAndProtectedSuggestions;
 }
+
+export function isLiteralXMLViewCompletion(
+  suggestion: UI5XMLViewCompletion
+): suggestion is LiteralXMLViewCompletion {
+  switch (suggestion.type) {
+    case "BooleanValueInXMLAttributeValueCompletion": {
+      // This is here so we'll get compilation errors if the type changes
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const _: LiteralXMLViewCompletion = suggestion;
+      return true;
+    }
+    default: {
+      return false;
+    }
+  }
+}
+export function isUI5NodeXMLViewCompletion(
+  suggestion: UI5XMLViewCompletion
+): suggestion is UI5NodeXMLViewCompletion {
+  // Suggestions are either literals or UI5 nodes
+  if (isLiteralXMLViewCompletion(suggestion)) {
+    return false;
+  }
+  // This is here so we'll get compilation errors if the type changes
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const _: UI5NodeXMLViewCompletion = suggestion;
+  return true;
```

Comment:
The logic here is only correct if you combine design time and runtime knowledge.
Anyone calling it from plain JavaScript (which they could because it is a public API)
is going to get unexpected results

Reply:
The compilation error mentioned is internal in this package. The logic will be correct no matter how you call it, but if we add a literal type this will make sure we don't forget to update this function.